template.go

```diff
@@ -5,49 +5,13 @@ import (
 	"path"
 )
 
-type Template interface {
-	// Name of this template.
-	Name() string
-	// Runtime for which this template applies.
-	Runtime() string
-	// Repository within which this template is contained.  Value is set to the
-	// currently effective name of the repository, which may vary. It is user-
-	// defined when the repository is added, and can be set to "default" when
-	// the client is loaded in single repo mode. I.e. not canonical.
-	Repository() string
-	// Fullname is a calculated field of [repo]/[name] used
-	// to uniquely reference a template which may share a name
-	// with one in another repository.
-	Fullname() string
-	// Write updates fields of Function f and writes project files to path pointed by f.Root.
-	Write(ctx context.Context, f *Function) error
-}
-
-type templateConfig struct {
-	// BuildConfig defines builders and buildpacks.  the denormalized view of
-	// members which can be defined per repo or per runtime first.
-	BuildConfig `yaml:",inline"`
-
-	// HealthEndpoints.  The denormalized view of members which can be defined
-	// first per repo or per runtime.
-	HealthEndpoints `yaml:"healthEndpoints,omitempty"`
-
-	// BuildEnvs defines environment variables related to the builders,
-	// this can be used to parameterize the builders
-	BuildEnvs []Env `yaml:"buildEnvs,omitempty"`
-
-	// Invocation defines invocation hints for a Functions which is created
-	// from this template prior to being materially modified.
-	Invocation Invocation `yaml:"invocation,omitempty"`
-}
-
 // template
 type template struct {
-	name       string
-	runtime    string
-	repository string
-	fs         Filesystem
-	manifest   templateConfig
+	name        string
+	runtime     string
+	repository  string
+	fs          Filesystem
+	templConfig templateConfig
```

Comment:
This is really a tiny nit, but couldn't this just be `config`. Since the struct is `template`, a `templateConfig` field name feels a little redundant.

Reply:
Renamed.